packages/map-deck/src/Common.ts

```diff
@@ -110,6 +111,41 @@ export class Common extends HTMLWidget {
                 this.syncMapbox(viewState);
             }
         });
+
+        this._mapgl.on("load", function() {
+            if (context.showBuildings()) {
+                const layers = context._mapgl.getStyle().layers;
+                let labelLayerId;
+                for (let i = 0; i < layers.length; i++) {
+                    if (layers[i].type === "symbol" && layers[i].layout["text-field"]) {
+                        labelLayerId = layers[i].id;
+                        break;
+                    }
+                }
+                context._mapgl.addLayer({
+                    "id": "3d-buildings",
+                    "source": "composite",
+                    "source-layer": "building",
+                    "filter": ["==", "extrude", "true"],
+                    "type": "fill-extrusion",
+                    "minzoom": 15,
+                    "paint": {
+                        "fill-extrusion-color": "#aaa",
+                        "fill-extrusion-height": [
+                            "interpolate", ["linear"], ["zoom"],
+                            15, 0,
+                            15.05, ["get", "height"]
+                        ],
+                        "fill-extrusion-base": [
+                            "interpolate", ["linear"], ["zoom"],
+                            15, 0,
+                            15.05, ["get", "min_height"]
+                        ],
+                        "fill-extrusion-opacity": .6
+                    }
+                }, labelLayerId);
+            }
```

Comment:
Code won't work if user disables / enables at runtime?

Reply:
Right, once the widget has rendered, this feature can't be toggled. Should we avoid features like this? Or recreate the widget when this feature is toggled?

Reply:
Recreate should be fine, its really only a design or user interaction type of thing.